### ISSUE TITLE
No clipping (only gradient noise)

### ISSUE THREAD

**Issue by jpgard** [pytorch/opacus#177]
Hi, I can't tell if this is support, or a feature request; please direct me elsewhere if needed.

How can I use a `PrivacyEngine` *without* gradient clipping (i.e., only add noise)?

I am not concerned about the theoretical privacy implications of this; it is related to a research experiment.

I've tried setting `max_grad_norm='inf'`, but this seems to lead to infinite gradients/nan outputs in my model after even a single gradient update; the training loop works fine when I set S to a "large" number, but this isn't the same as performing no clipping (and I assume there is at least a marginal additional overhead associated with clipping to even a large number).

**Comment by Darktex** [comment]
Hi! Yes, indeed this is something we do not currently support.

So, let's decompose this into two parts: 

1. Can I train efficiently with only adding noise on Opacus?
2. If I have no gradient clipping, how do I compute my (eps, delta) guarantee?

Let me answer 2 first, so it gives you context for my answer to 1 :) 

Remember that when you train with DP-SGD, you set a clipping threshold C and a noise multiplier sigma. When you do add noise, [that noise](https://github.com/pytorch/opacus/blob/master/opacus/privacy_engine.py#L579) has standard deviation **sigma C**, not just sigma. If C is infinite... I don't need to tell you the rest :P 

This is the reason why this use case is not supported in Opacus. Adding noise without clipping may very well be a fine privacy preserving technique, but it falls outside differential privacy, and we would not be able to compute a guarantee for it.

If you don't care about a privacy (eps, delta) guarantee, I would say just add noise from torch.normal to your gradients right before calling the optimizer, without using Opacus. It's a one-liner :)

**Comment by jpgard** [comment]
Thanks for the reply @Darktex !!
